annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java

```diff
@@ -671,24 +729,107 @@ private void validateTranscriptionWithFullTextResource(Body body)
 	    throw new PropertyValidationException(ParamValidationException.MESSAGE_MISSING_MANDATORY_FIELD,
 		    I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD, new String[] { "transcription.body.edmRights" });
-	}
+    }
 	validateEdmRights(body);
     }
+    
+    private void validateTagWithFullTextResource(Body body) throws ParamValidationException, PropertyValidationException {
+
+	// check mandatory fields
+
+	// check type
+	if (StringUtils.isBlank(body.getInternalType().toString()))
+	    throw new PropertyValidationException(ParamValidationException.MESSAGE_MISSING_MANDATORY_FIELD,
+		    I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD, new String[] { "tag.body.type" });
+
+	// check source
+	if (StringUtils.isBlank(body.getSource()))
+	    throw new PropertyValidationException(ParamValidationException.MESSAGE_MISSING_MANDATORY_FIELD,
+		    I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD, new String[] { "tag.body.source" });
+    }
 
     /**
      * @param body
      * @throws ParamValidationException
+     * @throws PropertyValidationException 
      */
-    private void validateVcardAddressBody(Body body) throws ParamValidationException {
+    private void validateVcardAddressBody(Body body) throws ParamValidationException, PropertyValidationException {
 
-	// check mandatory fields
-
-	// check type
+		// check mandatory fields
+    	validateSemanticTagVcardAddressBody(body);
+	
+		// check type
 	if (StringUtils.isBlank(body.getInternalType().toString())
-		|| !BodyInternalTypes.isVcardAddressTagBody(body.getInternalType()))
-	    throw new ParamValidationException(ParamValidationException.MESSAGE_MISSING_MANDATORY_FIELD,
-		    I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD,
-		    new String[] { "tag.body.type", body.getType().toString() });
+			|| !BodyInternalTypes.isVcardAddressTagBody(body.getInternalType()))
+		    throw new ParamValidationException(ParamValidationException.MESSAGE_MISSING_MANDATORY_FIELD,
+			    I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD,
+			    new String[] { "tag.body.type", body.getType().toString() });
+    }
+
+	/**
+     * Semantic tagging with vcard:Address type has mandatory fields:
+     * "Address", "streetAddress", "locality", "countryName"
+     * The "type" field should have value "Address"
+     * @param body
+     * @throws ParamValidationException
+	 * @throws PropertyValidationException 
+     */
+    private void validateSemanticTagVcardAddressBody(Body body) throws ParamValidationException, PropertyValidationException {
+		// check mandatory field address
+		if (!(body instanceof VcardAddressBody)) 
+		    throw new ParamValidationException(ParamValidationException.MESSAGE_MISSING_MANDATORY_FIELD,
+				    I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD,
+				    new String[] { "tag.body.class", body.getClass().toString() });
+
+		Address address = ((VcardAddressBody) body).getAddress();
+		
+		// check mandatory field streetAddress
+		if (Strings.isNullOrEmpty(address.getVcardStreetAddress()))
```

Comment:
Please remove the usage of Strings class and use StringUtils like the other methods

Reply:
Replaced by StringUtils